tree/tree/src/TTreeCache.cxx

```diff
@@ -614,10 +630,288 @@ Int_t TTreeCache::DropBranch(const char *bname, Bool_t subbranches /*= kFALSE*/)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Start of methods for the miss cache.
+////////////////////////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////////////////////////
+/// Enable / disable the miss cache.
+///
+/// The first time this is called on a TTreeCache object, the corresponding
+/// data structures will be allocated.  Subsequent enable / disables will
+/// simply turn the functionality on/off.
+void TTreeCache::SetOptimizeMisses(Bool_t opt) {
+
+   if (opt && !fMissCache) {
+      ResetMissCache();
+   }
+   fOptimizeMisses = opt;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Reset all the miss cache training.
+///
+/// The contents of the miss cache will be emptied as well as the list of
+/// branches used.
+void TTreeCache::ResetMissCache() {
+
+   fLastMiss = -1;
+   fFirstMiss = -1;
+
+   if (!fMissCache) {
+      fMissCache.reset(new MissCache());
+   }
+   fMissCache->clear();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// For the event currently being fetched into the miss cache, find the IO
+/// (offset / length tuple) to pull in the current basket for a given branch.
+///
+/// Returns:
+/// - IOPos describing the IO operation necessary for the basket on this branch
+/// - On failure, IOPos.length will be set to 0.
+TTreeCache::IOPos TTreeCache::FindBranchBasket(TBranch &b) {
```

Comment:
As this does not return a TBasket, a better (more consistent) name would be FindBranchBasketPos.

Also, I found (reading the code) that it would have helped me to make the entry number to be a explicit parameter (instead of being an implicit one on line 700).